backend/firestore/firestore.go

```diff
@@ -0,0 +1,174 @@
+package firestore
+
+import (
+	"context"
+	"time"
+
+	fs "cloud.google.com/go/firestore"
+	"github.com/kellegous/go/internal"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// NextID is the next numeric ID to use for auto-generated IDs
+type NextID struct {
+	ID uint32 `json:"id" firestore:"id"`
+}
+
+// Backend provides access to Google Firestore.
+type Backend struct {
+	db *fs.Client
+}
+
+// New instantiates a new Backend
+func New(path string) (*Backend, error) {
+	if path == "" {
+		path = getGoogleProject()
+	}
+	ctx := context.Background()
```

Comment:
`ctx` should probably be passed in here, right?

Reply:
It seems like `context.Background()` is the only reasonable value for the `New` call. Setting any timeout or deadline causes use of the client after that to fail with deadline exceeded. I guess I can just build that in `main.go`.